website/server/models/group.js

```diff
@@ -760,6 +770,168 @@ schema.methods.leave = async function leaveGroup (user, keep = 'keep-all') {
   return await Bluebird.all(promises);
 };
 
+schema.methods.updateTask = async function updateTask (taskToSync) {
+  let group = this;
+  let listOfPromises = [];
+
+  let usersToSync = await User.find({_id: { $in: taskToSync.assignedUsers} });
+
+  usersToSync.forEach(function syncAssignedUser (user) {
+    listOfPromises.push(group.syncTask(taskToSync, user));
+  });
+
+  return Bluebird.all(listOfPromises);
+};
+
+schema.methods.syncTask = async function groupSyncTask (taskToSync, user) {
+  let group = this;
+  let toSave = [];
+
+  if (taskToSync.assignedUsers.indexOf(user._id) === -1) {
+    taskToSync.assignedUsers.push(user._id);
+  }
+
+  // Sync tags
+  let userTags = user.tags;
+  let i = _.findIndex(userTags, {id: group._id});
+
+  if (i !== -1) {
+    if (userTags[i].name !== group.name) {
+      // update the name - it's been changed since
+      userTags[i].name = group.name;
+    }
+  } else {
+    userTags.push({
+      id: group._id,
+      name: group.name,
+    });
+  }
+
+  let findQuery = {
+    linkedTaskId: taskToSync._id,
+    userId: user._id,
+    'group.id': group._id,
+  };
+
+  let matchingTask = await Tasks.Task.findOne(findQuery).exec();
+
+  if (!matchingTask) { // If the task is new, create it
+    matchingTask = new Tasks[taskToSync.type](Tasks.Task.sanitize(syncableAttrs(taskToSync)));
+    matchingTask.group.id = taskToSync.group.id;
+    matchingTask.userId = user._id;
+    matchingTask.linkedTaskId = taskToSync._id;
+    user.tasksOrder[`${taskToSync.type}s`].push(matchingTask._id);
+  } else {
+    _.merge(matchingTask, syncableAttrs(taskToSync));
+    // Make sure the task is in user.tasksOrder
+    let orderList = user.tasksOrder[`${taskToSync.type}s`];
+    if (orderList.indexOf(matchingTask._id) === -1 && (matchingTask.type !== 'todo' || !matchingTask.completed)) orderList.push(matchingTask._id);
+  }
+
+  if (!matchingTask.notes) matchingTask.notes = taskToSync.notes; // don't override the notes, but provide it if not provided
+  if (matchingTask.tags.indexOf(group._id) === -1) matchingTask.tags.push(group._id); // add tag if missing
+
+  toSave.push(matchingTask.save(), taskToSync.save(), user.save());
+  return Bluebird.all(toSave);
+};
+
+schema.methods.unlinkTask = async function groupUnlinkTask (unlinkingTask, user, keep) {
+  let findQuery = {
+    linkedTaskId: unlinkingTask._id,
+    userId: user._id,
+  };
+
+  let assignedUserIndex = unlinkingTask.assignedUsers.indexOf(user._id);
+  unlinkingTask.assignedUsers.splice(assignedUserIndex, 1);
+
+  if (keep === 'keep-all') {
+    await Tasks.Task.update(findQuery, {
+      $unset: {'group.id': ''},
```

Comment:
Yeah I should have suggested to use $set before, sorry. The $unset was only for group.id but it's better to reset the entire object